core/change.go

```diff
@@ -100,79 +105,149 @@ func (change Change) Write(writer io.Writer) error {
-	var (
-		err  error
-		kind *KindConfig
-	)
-
-	if len(config.Components) > 0 {
-		compPrompt := promptui.Select{
-			Label: "Component",
-			Items: config.Components,
-			Stdin: stdinReader,
-		}
+func (change *Change) AskPrompts(config Config, stdinReader io.ReadCloser) error {
+	ctx := PromptContext{
+		config:      config,
+		stdinReader: stdinReader,
+		kind:        nil,
+	}
 
-		_, change.Component, err = compPrompt.Run()
-		if err != nil {
-			return err
-		}
+	err := change.promptForComponent(&ctx)
+
+	if err != nil {
+		return err
 	}
 
-	if len(config.Kinds) > 0 {
-		kindPrompt := promptui.Select{
-			Label: "Kind",
-			Items: config.Kinds,
-			Stdin: stdinReader,
-		}
+	err = change.promptForKind(&ctx)
 
-		_, change.Kind, err = kindPrompt.Run()
-		if err != nil {
-			return err
+	if err != nil {
+		return err
+	}
+
+	err = change.parseKind(&ctx)
+
+	if err != nil {
+		return err
+	}
+
+	err = change.promptForBody(&ctx)
+
+	if err != nil {
+		return err
+	}
+
+	return change.promptForUserChoices(&ctx)
+}
+
+func (change *Change) promptForComponent(ctx *PromptContext) error {
+	if len(ctx.config.Components) == 0 {
+		return nil
+	}
+
+	compPrompt := promptui.Select{
+		Label: "Component",
+		Items: ctx.config.Components,
+		Stdin: ctx.stdinReader,
+	}
+
+	var err error
+	_, change.Component, err = compPrompt.Run()
+
+	return err
+}
+
+func (change *Change) promptForKind(ctx *PromptContext) error {
+	if len(ctx.config.Kinds) == 0 || len(change.Kind) > 0 {
+		return nil
+	}
+
+	kindPrompt := promptui.Select{
+		Label: "Kind",
+		Items: ctx.config.Kinds,
+		Stdin: ctx.stdinReader,
+	}
+
+	var err error
+	_, change.Kind, err = kindPrompt.Run()
+
+	return err
+}
+
+func (change *Change) parseKind(ctx *PromptContext) error {
+	if len(change.Kind) == 0 {
+		return nil
+	}
+
+	for i := range ctx.config.Kinds {
+		kindConfig := &ctx.config.Kinds[i]
+
+		if kindConfig.Label == change.Kind {
+			ctx.kind = kindConfig
+			return nil
 		}
+	}
+
+	return fmt.Errorf("%w: %s", errInvalidKind, change.Kind)
+}
 
-		kind = kindFromLabel(config, change.Kind)
+func (change *Change) promptForBody(ctx *PromptContext) error {
+	if change.expectsNoBody(ctx) && len(change.Body) > 0 {
+		return errKindDoesNotAcceptBody
 	}
 
-	if kind == nil || !kind.SkipBody {
-		bodyPrompt := config.Body.CreatePrompt(stdinReader)
+	if change.expectsBody(ctx) && len(change.Body) == 0 {
+		bodyPrompt := ctx.config.Body.CreatePrompt(ctx.stdinReader)
+
+		var err error
 		change.Body, err = bodyPrompt.Run()
 
-		if err != nil {
-			return err
-		}
+		return err
+	}
+
+	if change.expectsBody(ctx) && len(change.Body) > 0 {
+		return ctx.config.Body.Validate(change.Body)
 	}
 
+	return nil
+}
+
+func (change *Change) expectsNoBody(ctx *PromptContext) bool {
+	return ctx.kind != nil && ctx.kind.SkipBody
+}
+
+func (change *Change) expectsBody(ctx *PromptContext) bool {
+	return ctx.kind == nil || !ctx.kind.SkipBody
+}
```

Comment:
I feel like these would make more since being part of the context instead of the change.
```suggestion
func (ctx *PromptContext) expectsNoBody() bool {
	return ctx.kind != nil && ctx.kind.SkipBody
}

func (ctx *PromptContext) expectsBody() bool {
	return ctx.kind == nil || !ctx.kind.SkipBody
}
```